rotkehlchen/data_import/importers/kucoin.py

```diff
@@ -0,0 +1,113 @@
+import csv
+from enum import Enum, auto
+from pathlib import Path
+from typing import Any
+
+from rotkehlchen.assets.converters import asset_from_kucoin
+from rotkehlchen.data_import.utils import BaseExchangeImporter
+from rotkehlchen.db.drivers.gevent import DBCursor
+from rotkehlchen.errors.misc import InputError
+from rotkehlchen.errors.serialization import DeserializationError
+from rotkehlchen.exchanges.data_structures import Trade
+from rotkehlchen.exchanges.utils import get_key_if_has_val
+from rotkehlchen.history.deserialization import deserialize_price
+from rotkehlchen.serialization.deserialize import (
+    deserialize_asset_amount,
+    deserialize_fee,
+    deserialize_timestamp_from_date,
+)
+from rotkehlchen.types import Location, TradeType
+
+
+class KucoinFileType(Enum):
+    """The type of the file we are parsing"""
+    TRADES = auto()  # default header format
+    TRADES_OLD = auto()  # old header format
+
+
+class KucoinImporter(BaseExchangeImporter):
+    @staticmethod
+    def _consume_trades(
+            csv_row: dict[str, str],
+            file_type: KucoinFileType,
+            timestamp_format: str = '%Y-%m-%d %H:%M:%S',
+    ) -> Trade:
+        """
+        Consume entries of kucoin order history csv
+
+        May raise:
+        - KeyError
+        - DeserializationError
+        """
+        if file_type == KucoinFileType.TRADES:
+            date = csv_row['tradeCreatedAt']
+            base, quote = csv_row['symbol'].split('-')
+            trade_type = TradeType.BUY if csv_row['side'] == 'buy' else TradeType.SELL
+            amount = csv_row['size']
+            rate = csv_row['price']
+            fee = get_key_if_has_val(csv_row, 'fee')
+            fee_currency = get_key_if_has_val(csv_row, 'feeCurrency')
+        else:
+            date = csv_row['Time']
+            base, quote = csv_row['Pair'].split('/')
+            trade_type = TradeType.BUY if csv_row['Side'] == 'Buy' else TradeType.SELL
```

Comment:
could you do something like

```suggestion
            trade_type_raw = csv_row['Side'].lower()
```

for both and then just compare 

```suggestion
                        trade_type = TradeType.BUY if csv_row['Side'] == 'buy' else TradeType.SELL
```

this way you avoid duplicating all the logic